src/model/ActionSet.cs

```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
 
       return state;
     };
+
+    public static SpecifiedAction<int, int, int, int> PerformRandomAction = (state, onSelf, action, targetAmount, power, remainArguments) =>
+    {
+      if (onSelf == 0)
+      {
+        var player = state.Players[0].Id != state.CurrentPlayer.Id ? state.Players[0] : state.Players[1];
+        state = PerformRandomActionOnPlayer(state, player, action, targetAmount, power);
+      }
+      else
+      {
+        var player = state.CurrentPlayer;
+        state = PerformRandomActionOnPlayer(state, player, action, targetAmount, power);
+      }
+      return state;
+    };
+
+    private static GameState PerformRandomActionOnPlayer(GameState state, Player player, int action, int targetAmount, int power)
```

Comment:
Need renaming
This method performs provided action on a random card
But your naming means that "i need to perform random action on a card"
Suggestion: 
- `PerformRandomAction` -> `PerformActionOnCard`
- `PerformRandomActionOnPlayer` -> `PerformActionOnCardWithPlayer`